app/helpers.php

```diff
@@ -93,3 +95,40 @@ function getRtlCss($path)
         return implode('/', $path).'/'.$filename.'.rtl.css';
     }
 }
+
+if (! function_exists('redirectToHome')) {
+    /**
+     * Redirect to the "home" page depending on authentication/authorization status.
+     *
+     * @param $flashMsg
+     * @param $$arrData
+     */
+    function redirectToHome($flashMsg = null, $arrData = null)
+    {
+        if (access()->allow('view-backend')) {
+            return redirect()->route('admin.dashboard')->withFlashDanger($flashMsg, $arrData);
+        } elseif (Auth::check()) {
+            return redirect()->route('frontend.user.dashboard')->withFlashDanger($flashMsg, $arrData);
+        }
+
+        return redirect()->route('frontend.index')->withFlashDanger($flashMsg, $arrData);
+    }
```

Comment:
I'm not sure what @rappasoft would prefer, but since this is one change request, it would be nice if it was squashed into a single commit.